app/sessionStore.js

```diff
@@ -837,11 +837,10 @@ module.exports.runPreMigrations = (data) => {
 }
 
 module.exports.runPostMigrations = (immutableData) => {
-  const globalFpSetting = immutableData.getIn(['settings', 'privacy.block-canvas-fingerprinting'])
-  // fingerprinting protection migration
-  if (typeof globalFpSetting !== 'boolean') {
+  if (immutableData.get('fingerprintingProtectionAll')) {
```

Comment:
If more logic is added to `runPostMigrations` (ex: after FP migration), this condition might be triggered causing an immediate exit. Can you create a new method- like `fingerprintProtectionMigration` and move this logic there? Then the new code would be like:
```
module.exports.runPostMigrations = (immutableData) => {
  fingerprintProtectionMigration()
  // other post migrations can go here
}
```